I ran the Kubernetes tests locally yesterday, and they all passed. However, after rebasing today, they consistently fail.  
It seems that triggering a `dagRun` results in a 404 error:  
https://github.com/apache/airflow/blob/2ea7aed755006cfff3dedd30b5654e8e2534e197/kubernetes_tests/test_base.py#L270  
```
self = <requests.adapters.HTTPAdapter object at 0x119d835b0>, request = <PreparedRequest [POST]>, stream = False, timeout = Timeout(connect=None, read=None, total=None), verify = True, cert = None, proxies = OrderedDict()

# ...
    
        except MaxRetryError as e:
            if isinstance(e.reason, ConnectTimeoutError):
                # TODO: Remove this in 3.0.0: see #2811
                if not isinstance(e.reason, NewConnectionError):
                    raise ConnectTimeout(e, request=request)
    
            if isinstance(e.reason, ResponseError):
>               raise RetryError(e, request=request)
E               requests.exceptions.RetryError: HTTPConnectionPool(host='localhost', port=14476): Max retries exceeded with url: /public/dags/example_kubernetes_executor/dagRuns (Caused by ResponseError('too many 404 error responses'))
```

Tracing the issue on the API side, it looks like the DAG must be **active** to be triggered:  
https://github.com/apache/airflow/blob/2ea7aed755006cfff3dedd30b5654e8e2534e197/airflow/api_fastapi/core_api/routes/public/dag_run.py#L349-L352  

However, in the **patch DAG** endpoint, only the `is_paused` field can be modified, while the `is_active` field cannot be changed:  
https://github.com/apache/airflow/blob/2ea7aed755006cfff3dedd30b5654e8e2534e197/airflow/api_fastapi/core_api/routes/public/dags.py#L218-L228

--- a/kubernetes_tests/test_base.py
+++ b/kubernetes_tests/test_base.py
@@ -179,7 +179,7 @@ class BaseK8STest:
         session = requests.Session()
         session.headers.update({"Authorization": f"Bearer {jwt_token}"})
         retries = Retry(
-            total=3,
+            total=5,
             backoff_factor=10,
             status_forcelist=[404],
             allowed_methods=Retry.DEFAULT_ALLOWED_METHODS | frozenset(["PATCH", "POST"]),

--- a/kubernetes_tests/test_base.py
+++ b/kubernetes_tests/test_base.py
@@ -63,6 +63,11 @@ class BaseK8STest:
         self.set_api_server_base_url_config()
         self.rollout_restart_deployment("airflow-api-server")
         self.ensure_deployment_health("airflow-api-server")
+
+        # Sometimes the DAGs are not serialized yet, will cause the trigger dagRun API to get 404 error
+        # so we need to do it manually
+        self.dags_reserialize()
+
         # Replacement for unittests.TestCase.id()
         self.test_id = f"{request.node.cls.__name__}_{request.node.name}"
         self.session = self._get_session_with_retries()
@@ -128,7 +133,7 @@ class BaseK8STest:
             check_call(["kubectl", "delete", "pod", names[0]])
 
     @staticmethod
-    def _get_jwt_token(session: requests.Session, username: str, password: str) -> str:
+    def _get_jwt_token(username: str, password: str) -> str:
         """Get the JWT token for the given username and password.
 
         Note: API server is still using FAB Auth Manager.
@@ -148,6 +153,7 @@ class BaseK8STest:
         :return: The JWT token
         """
         # get csrf token from login page
+        session = requests.Session()
         get_login_form_response = session.get(f"http://{KUBERNETES_HOST_PORT}/auth/login")
         # input id="csrf_token"
         csrf_token = re.search(
@@ -180,8 +186,8 @@ class BaseK8STest:
         return jwt_token
 
     def _get_session_with_retries(self):
+        jwt_token = self._get_jwt_token("admin", "admin")
         session = requests.Session()
-        jwt_token = self._get_jwt_token(session, "admin", "admin")
         session.headers.update({"Authorization": f"Bearer {jwt_token}"})
         retries = Retry(
             total=3,
@@ -316,6 +322,23 @@ class BaseK8STest:
                 "merge",
                 "-p",
                 f'{{"data": {{"{configmap_key}": "{self._parse_airflow_cfg_dict_as_escaped_toml(airflow_cfg_dict)}"}}}}',
+            ]
+        )
+
+    @staticmethod
+    def dags_reserialize():
+        """Reserialize the DAGs using the airflow CLI in the scheduler pod."""
+        check_call(
+            [
+                "kubectl",
+                "-n",
+                "airflow",
+                "exec",
+                "deployment/airflow-scheduler",
+                "--",
+                "airflow",
+                "dags",
+                "reserialize",
             ]
         )
 

--- a/kubernetes_tests/test_base.py
+++ b/kubernetes_tests/test_base.py
@@ -25,6 +25,7 @@ import time
 from datetime import datetime, timezone
 from pathlib import Path
 from subprocess import check_call, check_output
+from urllib.parse import parse_qs, urlparse
 
 import pytest
 import requests
@@ -126,9 +127,62 @@ class BaseK8STest:
         if names:
             check_call(["kubectl", "delete", "pod", names[0]])
 
+    @staticmethod
+    def _get_jwt_token(session: requests.Session, username: str, password: str) -> str:
+        """Get the JWT token for the given username and password.
+
+        Note: API server is still using FAB Auth Manager.
+
+        Steps:
+        1. Get the login page to get the csrf token
+            - The csrf token is in the hidden input field with id "csrf_token"
+        2. Login with the username and password
+            - Must use the same session to keep the csrf token session
+        3. Extract the JWT token from the redirect url
+            - Expected to have a connection error
+            - The redirect url should have the JWT token as a query parameter
+
+        :param session: The session to use for the request
+        :param username: The username to use for the login
+        :param password: The password to use for the login
+        :return: The JWT token
+        """
+        # get csrf token from login page
+        get_login_form_response = session.get(f"http://{KUBERNETES_HOST_PORT}/auth/login")
+        # input id="csrf_token"
+        csrf_token = re.search(
+            r'<input id="csrf_token" name="csrf_token" type="hidden" value="(.+?)">',
+            get_login_form_response.text,
+        )
+        assert csrf_token, "Failed to get csrf token from login page"
+        csrf_token_str = csrf_token.group(1)
+        assert csrf_token_str, "Failed to get csrf token from login page"
+        try:
+            # login with form data
+            session.post(
+                f"http://{KUBERNETES_HOST_PORT}/auth/login",
+                data={"username": username, "password": password, "csrf_token": csrf_token_str},
+            )
+        except requests.exceptions.ConnectionError as e:
+            # expected to have a connection error
+            # currently, the login page redirects to http://localhost:8080/?token=... with status code 308
+            # but the KUBERNETES_HOST_PORT is *not* localhost:8080
+            # TODO: remove this try/except block when the redirect url is fixed
+            redirect_url = e.request.url if e.request else None
+            # ensure redirect_url is a string
+            redirect_url_str = str(redirect_url) if redirect_url is not None else ""
+            assert "/?token" in redirect_url_str, f"Login failed with redirect url {redirect_url_str}"
+            parsed_url = urlparse(redirect_url_str)
+            query_params = parse_qs(str(parsed_url.query))
+            jwt_token_list = query_params.get("token")
+            jwt_token = jwt_token_list[0] if jwt_token_list else None
+        assert jwt_token, f"Failed to get JWT token from redirect url {redirect_url_str}"
+        return jwt_token
+
     def _get_session_with_retries(self):
         session = requests.Session()
-        session.auth = ("admin", "admin")
+        jwt_token = self._get_jwt_token(session, "admin", "admin")
+        session.headers.update({"Authorization": f"Bearer {jwt_token}"})
         retries = Retry(
             total=3,
             backoff_factor=10,

--- a/kubernetes_tests/test_kubernetes_executor.py
+++ b/kubernetes_tests/test_kubernetes_executor.py
@@ -50,7 +50,7 @@ class TestKubernetesExecutor(BaseK8STest):
             timeout=300,
         )
 
-    @pytest.mark.execution_timeout(400)
+    @pytest.mark.execution_timeout(500)
     def test_integration_run_dag_with_scheduler_failure(self):
         dag_id = "example_kubernetes_executor"
 

